source/math/FloatRotation3D.ooc

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+import math
+import FloatExtension
+import FloatPoint3D
+import FloatTransform2D
+
+FloatRotation3D: cover {
+	x, y, z: Float
+	init: func@ ~full(=x, =y, =z)
+	init: func@ ~default { this init(0.0f, 0.0f, 0.0f) }
+	init: func@ ~fromPoint(point: FloatPoint3D) { this init(point x, point y, point z) }
+	getTransform: func(k: Float) -> FloatTransform2D {
+			rx := FloatTransform2D createXRotation(this x, k)
+			ry := FloatTransform2D createYRotation(this y, k)
+			rz := FloatTransform2D createZRotation(this z)
+			return rz * rx * ry // Yaw -> Pitch -> Roll
```

Comment:
Indentation...

Also, have you verified this is correct? A test would be nice.